data-prepper-api/src/main/java/org/opensearch/dataprepper/model/codec/InputCodec.java

```diff
@@ -3,25 +3,20 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.opensearch.dataprepper.plugins.source.codec;
+package org.opensearch.dataprepper.model.codec;
 
 import org.opensearch.dataprepper.model.event.Event;
 import org.opensearch.dataprepper.model.record.Record;
-
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.function.Consumer;
 
-/**
- * A codec parsing data through an input stream. Each implementation of this class should
- * support parsing a specific type or format of data. See sub-classes for examples.
- */
-public interface Codec {
+public interface InputCodec {
     /**
      * Parses an {@link InputStream}. Implementors should call the {@link Consumer} for each
      * {@link Record} loaded from the {@link InputStream}.
      *
-     * @param inputStream   The input stream for the S3 object
+     * @param inputStream   The input stream for the source plugin(e.g. S3, Http, RssFeed etc) object
      * @param eventConsumer The consumer which handles each event from the stream
      */
     void parse(InputStream inputStream, Consumer<Record<Event>> eventConsumer) throws IOException;
```

Comment:
Because the function signature has `throws IOException`, the javadoc also should have a line `@throws IOException throws IOException when <put the condition when the exception is thrown here>`

Reply:
Done